### ISSUE TITLE
Manga panel zoom: black rectangles and no zoom

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#7961]
* KOReader version: 2021-06-24
* Device: Kindle PW3

#### Issue
Some of the panels are either zoomed to a black rectangle (sometimes half-black, half-image) or cannot be zoomed at all, although they have well-defined borders without weird stuff overlapping them.

#### Steps to reproduce
Open this sample file (rename `.zip` to `.cbz`):
[minamoto-0112-0003-kcc.zip](https://github.com/koreader/koreader/files/6797868/minamoto-0112-0003-kcc.zip)
Long-tap panels one by one.

Here is what I get (top-to-bottom, left-to-right):
Row 1 (3 panels): half-black rectangle zoom; good zoom; black rectangle zoom;
Row 2 (1 panel): no zoom
Row 3 (3 panels): no zoom on either
Row 4 (2 panels): no zoom; good zoom.

Without `Allow panel zoom` in Top menu it defaults to text selection.
Not a big deal, since I can rotate landscape or zoom manually (however, `Zoom factor` also has some inconsistencies: #7962).

**Comment by Galunid** [comment]
I'll see what's wrong, I can reproduce and it looks pretty weird

**Comment by Galunid** [comment]
There is non-zero change #1407 fixed this ^_^ (once it's bumped)

**Comment by Frenzie** [comment]
I don't think it's affected. Sorry. :-)

**Comment by Galunid** [comment]
@feress737 Did you try this with the attached file specifically? In general I'm pretty sure panel zoom works as expected, it's the attached file that breaks it. I'm not exactly sure why since it doesn't happen with the manga I usually read.

**Comment by feress737** [comment]
@Galunid No, I noticed the effect with the files on my device (Tolino Epos 1). In the meantime I have found that the page refresh has no effect and the image in the panel zoom view is still often black. After such a black panel is displayed, the back button no longer works.

**Comment by Galunid** [comment]
Closed in #12296

**Comment by NiLuJe** [comment]
The missed panel detection is still an issue, only the actual rendering issues when a panel rectangle is properly detected has been fixed ;).

**Comment by Galunid** [comment]
See my comment https://github.com/koreader/koreader/pull/12296#issuecomment-2269766417 explaining why panels are not detected here ;)

**Comment by NiLuJe** [comment]
> See my comment [#12296 (comment)](https://github.com/koreader/koreader/pull/12296#issuecomment-2269766417) explaining why panels are not detected here ;)

Oh, cool, I'd missed that.

Seems to be limited to relatively narrow panels in my own very meager experience, FWIW.

**Comment by mergen3107** [comment]
Guys, I forgot how to panel zoom in manga 💀
I long tap panels but all I get is interface with the whole page in Picture Viewer where I can zoom and all. 

"Panel zoom" is enabled in top menu.  Am I missing something?

I wanted to try the fixes for ratios

**Comment by Frenzie** [comment]
Maybe the file in question has very hard to detect panels?
